slick.grid.js

```diff
@@ -3108,30 +3122,44 @@ if (typeof Slick === "undefined") {
       return options;
     }
 
+    function updateOptions(suppressRender, suppressColumnSet, suppressSetOverflow) {
+      prepareForOptionsChange();
+      trigger(self.onUpdateOptions, { "options": options });
+      internal_setOptions(suppressRender, suppressColumnSet, suppressSetOverflow);
+    }
+
     function setOptions(args, suppressRender, suppressColumnSet, suppressSetOverflow) {
+      prepareForOptionsChange();
+
+      var originalOptions = utils.extend(true, {}, options);
+      options = utils.extend(options, args);
+      trigger(self.onSetOptions, { "optionsBefore": originalOptions, "optionsAfter": options });
+
+      internal_setOptions(suppressRender, suppressColumnSet, suppressSetOverflow);
+    }
+
+    function prepareForOptionsChange() {
       if (!getEditorLock().commitCurrentEdit()) {
         return;
       }
 
       makeActiveCellNormal();
 
-      if (args.showColumnHeader !== undefined) {
-        setColumnHeaderVisibility(args.showColumnHeader);
-      }
-
-      if (options.enableAddRow !== args.enableAddRow) {
+      //if (typeof args.enableAddRow !== 'undefined' && options.enableAddRow !== args.enableAddRow) {
```

Comment:
does this code really have to be commented out? isn't that going to break something?

Reply:
I don't think so, I decided it was actually redundant, but I'd have to double check. We no longer have the before and after state of the options.